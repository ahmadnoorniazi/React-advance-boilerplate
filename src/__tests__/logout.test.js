// these should normally be in your jest setupTestFrameworkScriptFile
import "jest-dom/extend-expect";
import "react-testing-library/cleanup-after-each";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { fireEvent, cleanup, render } from "react-testing-library";
import reducer from "../reducers";
import React from "react";
import Chat from "../Chat";

const key = "state";
const fakeData = {
  name: "ahmad",
  email: "ahmad@gmail.com"
};
beforeEach(() => {
  window.localStorage.setItem(key, JSON.stringify([fakeData]));
});

test("should test for logout ", () => {
  const store = createStore(reducer);
  window.HTMLElement.prototype.scrollIntoView = function() {};
  window.alert = function() {};
  const fakeHistory = { push: jest.fn() };
  const { container, getByDisplayValue } = render(
    <Provider store={store}>
      <Chat history={fakeHistory} />
    </Provider>
  );
  const logoutButton = getByDisplayValue("logout");
  expect(JSON.parse(window.localStorage.getItem("state"))).toEqual([fakeData]);
  fireEvent.click(logoutButton);
  expect(logoutButton.type).toBe("button");
  expect(fakeHistory.push).toHaveBeenCalledTimes(1);
  expect(window.localStorage.getItem("state")).toBe(null);
  expect(fakeHistory.push).toHaveBeenCalledWith("./login");
  expect(container.firstChild).toMatchSnapshot();
});
afterEach(cleanup);
