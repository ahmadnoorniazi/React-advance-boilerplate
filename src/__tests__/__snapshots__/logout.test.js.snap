// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test for sign up  1`] = `
<div>
  <div
    class="heading"
  >
    <h3
      data-testid="chat-title"
    >
      Welcome to My Chat App
    </h3>
    <div
      class="log-out"
    >
      <input
        class="logOut-btn"
        type="button"
        value="logout"
      />
    </div>
  </div>
  <article
    class="Message"
  >
    <div
      class="Message--main"
      style="overflow: auto; height: 450px;"
    >
      <ul
        class="Uol"
      >
        <li>
          <h4
            class="Message--user"
          >
            ahmad
          </h4>
          <p
            class="Message--content"
          />
        </li>
      </ul>
      <div />
    </div>
  </article>
  <input
    class="message-field"
    placeholder="Type your messege and hit ENTER"
    type="text"
    value=""
  />
</div>
`;
